co2_sensor_32u4/32u4_co2.ino

```diff
@@ -0,0 +1,152 @@
+/*
+ * Modified by : Bee Kaan 
+ * Last updated : 22/05/2020
+ * Original Code from https://electronics.stackexchange.com/questions/262473/mh-z19-co2-sensor-giving-diferent-values-using-uart-and-pwm
+ * Datasheet Link for reference : https://gitlab.op-bit.nz/BIT/Project/Internet-Of-Things/nodes/blob/CO2/MH-Z19B_CO2_Sensor.pdf
+ * Project : C02 monitor
+ * Description : This code is designed to calculate indoor level of c02 gas concentration with 5 seconds between each reading.  
+ *                The Parts Per Million (PPM) for this project is obtained from the PWM output.  
+ *                Therefore, the calculation is to generate data with 2000 ppm detection range based on information from datasheet.                  
+ *                The project is created on Adafruit Feather 32u4 and MH-Z19B c02 sensor.  
+ * More information on C02 level : https://www.kane.co.uk/knowledge-centre/what-are-safe-levels-of-co-and-co2-in-rooms
+ *              
+ */
+#include <TinyLoRa.h>
+#include <SPI.h>
+#include "MHZ19.h"
+
+#define BAUDRATE 9600
+
+MHZ19 myMHZ19;
+
+//global variables for temperature and Humidity
+float temperature = 0;
+float CO2 = 0;
+
+
+// Network Session Key (MSB)
+uint8_t NwkSkey[16] = { FILLMEIN };
+
+// Application Session Key (MSB)
+uint8_t AppSkey[16] = { FILLMEIN };
+
+
+// Device Address (MSB)
+uint8_t DevAddr[4] = { FILLMEIN };
+
+
+/************************** Magic Begins Here ***********************************/
+
+
+//identify response code
+unsigned char response[9];
+
+unsigned char payload[5];
+
+
+// Pinout for Adafruit Feather 32u4 LoRa
+TinyLoRa lora = TinyLoRa(7, 8, 4);
+
+void setup() {
+  Serial.begin(9600);  //display on serial monitor
+  while (!Serial)
+    ;
+
+  Serial1.begin(9600);
+
+  myMHZ19.begin(Serial1);    // Initialize the sensor with hardware serial port 1
+  
+  delay(20000000); // Warmup delay before calibration
```

Comment:
Was there a requirement for the delay? Could that be added as a comment just so we know not to remove it?

Reply:
This is to allow the CO2 sensor to warm up before calibrating and collecting data